File: tests/my_runner.py
#!/usr/bin/env python3
"""Run a qunit test with selenium/phantomjs

Usage:
  my_runner.py <path_to_tests.html>

Options:
  -h --help                                 Show this screen.

"""
from docopt import docopt
from selenium import webdriver
import os

def main(path):
    browser = webdriver.PhantomJS()
    try:
        browser.get('file:///' + path)
        body = browser.find_element_by_tag_name('body')
        #print(browser.page_source)
        #print(body.text)
        print()
        overall_result = browser.find_element_by_id('qunit-testresult')
        headline = overall_result.text.split('\n')[1]
        print(headline)
        test_lis = browser.find_elements_by_css_selector('#qunit-tests li')
        tests = [
            li for li in test_lis
            if 'qunit-test-output' in li.get_attribute('id')
        ]
        for ix, test in enumerate(tests):
            result_text = test.text.split('\n')[0]
            result_text = result_text.replace('Rerun', '')
            print('{}. {}'.format(ix + 1, result_text))

    finally:
        browser.quit()


if __name__ == '__main__':
    arguments = docopt(__doc__)
    path = os.path.abspath(arguments['<path_to_tests.html>'])
    main(path)

File: tests/my_runner.js
var system = require('system');
// var env = system.env;
// Object.keys(env).forEach(function(key) {
//     console.log(key + '=' + env[key]);
// });

var path = system.env.PWD + '/' + system.args[1];
console.log(path);

if (!system.args[1]){
    console.log('Pass path to test file as second arg');
    phantom.exit();
}

var page = require('webpage').create();
page.open('file://' + path, function () {
    setTimeout(function() {
        var output = page.evaluate( function () {
            var results = '';

            var headline = $('#qunit-testresult').text().split('.')[1];
            results += headline + '\n';

            var counter = 0;
            $('#qunit-tests li').each(function() {
                var li = $(this);
                if (li.prop('id').indexOf('qunit-test-output') !== -1){
                    counter += 1;
                    var resultLine = '';
                    resultLine += counter + '. ';
                    resultLine += li.find('.test-name').text();
                    resultLine += ' ' + li.find('.counts').text();
                    resultLine = resultLine.replace('Rerun', '');
                    var fails = li.find('.fail');
                    if (fails.text()) {
                        resultLine += '\n';
                        resultLine += fails.find('tr.test-expected').text();
                        resultLine += '\n';
                        resultLine += fails.find('tr.test-actual').text();
                        resultLine += '\n';
                        resultLine += fails.find('tr.test-diff').text();
                        resultLine += '\n';
                        resultLine += fails.find('tr.test-source').text();
                    }
                    results += resultLine + '\n';
                }
            });

            return results;

            //return document.getElementById('qunit-testresult').textContent;
        });
        console.log(output);
        phantom.exit();

    }, 100);
});

